yt_dlp/utils.py

```diff
@@ -5876,6 +5876,153 @@ def orderedSet_from_options(options, alias_dict, *, use_regex=False, start=None)
     return orderedSet(requested)
 
 
+class InitializationFragmentError(YoutubeDLError):
+    def __init__(self):
+        super().__init__(
+            'Initialization fragment found after media fragments, unable to download')
+
+
+class HlsMediaManifest:
+
+    def __init__(self, manifest, manifest_url):
```

Comment:
shouldnt this be
```suggestion
    def __init__(self, manifest, manifest_url=None):
```

Reply:
I think it needs to be required because in some cases it will do urljoin

```python
frag_url = (
               line
               if re.match(r'^https?://', line)
               else urllib.parse.urljoin(self.manifest_url, line))
```